### PR TITLE
Fixes #2978

### LINKED ISSUE #2978: Auto-Disenchanter has strict slot order
## :round_pushpin: Description (REQUIRED)

The auto disenchanter does not work if the book is on the left (first) slot

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

1. Place auto disenchanter
2. Insert an book into the left (first) slot 
3. Insert an enchanted item into the right (second) slot
4. Nothing happens
If the enchanted item is placed in the right (second) slot and the book in the left (first), the auto disenchanter works.

## :bulb: Expected behavior (REQUIRED)

The auto disenchanter will work

## :scroll: Server Log

N/A

## :open_file_folder: /error-reports/ Folder

N/A

## :compass: Environment (REQUIRED)

image

### PR DESCRIPTION
**Pull request by variananora**
## Description

To fix strict order slot on Auto Disenchanter

## Proposed changes

Fix is stolen from #2982

## Related Issues (if applicable)

Fixes #2978

## Video Demonstration and Testing

https://youtu.be/akOvhaHzmzM

## Checklist

- [X] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [X] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java [hunk 1]
```diff
@@ -129,21 +127,15 @@ private void transferEnchantments(ItemStack item, ItemStack book, Map<Enchantmen
...
 
-        EnchantmentStorageMeta meta = (EnchantmentStorageMeta) book.getItemMeta();
-
         for (Map.Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
             item.removeEnchantment(entry.getKey());
-            meta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
         }
 
-        book.setItemMeta(meta);
+        book.addUnsafeEnchantments(enchantments);
```
Comment (Mooy1) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java:134]: You need the EnchantmentStorageMeta not the book's enchants

Comment (variananora) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java]: Reverted to the old ones.

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java [hunk 2]
```diff
@@ -55,10 +55,10 @@ public ItemStack getProgressBar() {
     @Override
     protected MachineRecipe findNextRecipe(BlockMenu menu) {
         for (int slot : getInputSlots()) {
-            ItemStack item = menu.getItemInSlot(slot);
+            ItemStack item = menu.getItemInSlot(slot == getInputSlots()[0] ? getInputSlots()[1] : getInputSlots()[0]);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java:58]: Why is item and secondItem swapped here?

Comment (variananora) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java:58]: No reason actually.... Just to follow AutoEnchanter.java styling.

**Review by svr333 — DISMISSED** [review]
Looks fine

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java (hunk 2 above)
Comment (variananora) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java]: Reverted.

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Comment by WalshyDev** [comment]
Cookie add to changelog <3

**Comment by variananora** [comment]
I should just add to the Changelog myself if I make another PR...

**Pull request merged**